vtm/src/org/oscim/renderer/OffscreenRenderer.java

```diff
@@ -192,15 +192,16 @@ public void update(GLViewport viewport) {
 
     @Override
     public void render(GLViewport viewport) {
-        gl.bindFramebuffer(GL.FRAMEBUFFER, fb);
-        gl.viewport(0, 0, texW, texH);
+        int fbTmp = GLState.getFramebuffer();
+        GLState.bindFramebuffer(fb);
+        GLState.viewport(texW, texH);
         gl.depthMask(true);
         GLState.setClearColor(mClearColor);
         gl.clear(GL.DEPTH_BUFFER_BIT | GL.COLOR_BUFFER_BIT);
 
         mRenderer.render(viewport);
 
-        gl.bindFramebuffer(GL.FRAMEBUFFER, 0);
+        GLState.bindFramebuffer(fbTmp);
```

Comment:
Why change from `0`?

Reply:
It has not changed from 0 in default cases.
Reason is, if having a renderer on top which uses a framebuffer, too, this framebuffer is lost and afterlying renderer won't render to its framebuffer.
I only wanted to think foreward, if someone plans another renderer after Offscreenrenderer.
I made the analog in ShadowRenderer, where I write to the framebuffer from lights perpective, but possible OffscreenRenderer comes on top, which uses another framebuffer.

Can leave as there aren't plans for other renderers, but this may can cause more trouble in the future.

Reply:
Generally we should try move away from many static references, could the reason for #194.

Reply:
Yes, but that's not the point here (?)

Reply:
Whole `GLState` class must be converted to dynamic object, but that not solves this particular case.

Reply:
Probably, though never invested time to that, as library currently works fine and that's most important.

Reply:
Also any change should clearly improve or fix something, or else there is not real reason to mess with something that works fine.